### PR TITLE
[WIP] Implement new look-and-feel for user registration page

### PR DESCRIPTION
**Pull request by toolness**
This implements the new look-and-feel for the user registration page from [claudioccm/noi-ui](https://github.com/claudioccm/noi-ui).

Notes/concerns:
* **We currently require a full form submission in order to do any kind of form validation.** While I think we can/should improve on this in the future, for now this means:
  * We can't really make use of the `m-valid` style right now.
  * We *can* use the `m-invalid` style, but note that for now we can't reliably remove this style, so the user might be confused if e.g. they change the value of the password field but the styling still indicates that something's wrong with it.
  * We can't show any error messages until the user has submitted the whole form, which makes the new UI's style of only displaying one message at a time at the bottom of the display problematic. (That style of alert is great when we can show users errors as they fill out the form, but it's not so great when we have to show the user multiple errors after they've submitted the form.)
* ~~I have concerns about using `text-transform: uppercase` for text fields that are case-retentive. For instance, a user could mis-type their name as "JOhn" but not realize that there is a typo until they see the mixed-case version of their name elsewhere on the site.~~
* There doesn't seem to be any kind of outline or other styling on the submit button to indicate that it's focused. I can't find an `outline: none` for this button anywhere in the CSS, but having no separate styling to indicate focus is [not great for accessibility](http://www.outlinenone.com/).
* ~~Currently, all security-related pages have links to "login", "register", and "forgot password" at the bottom, but because I don't know how we want that to be styled, they're not visible yet on the registration page.~~
* The new base template is including *both* [Google Material Icons](https://www.google.com/design/icons/) *and* [Font Awesome](http://fortawesome.github.io/Font-Awesome/); are we really using both? (I definitely see the material icons in claudio's HTML source, but haven't seen any FA yet.)
* I'd like to figure out how to make TypeKit work for development machines, because currently typekit is denying my dev deployment with 403's, so I can't tell what the site is ideally supposed to look like. I suspect this has something to do with the fact that I can't access my dev deployment at 127.0.0.1/localhost because docker.
* The close button on flash messages is currently a `<div>`, but it should really be a `<button>` for full accessibility (an alternative is for us to add a bunch of stuff like keyboard event handlers and `tabindex` attributes, but that's quite a hassle). The same goes for any other elements on the page that have the functionality of buttons, even if they don't look like them.

### CODE REVIEW

**Comment by claudioccm** [comment]
I agree with your concerns. Use whatever styles you see fit. And we can
take out the text-transform.

I will revisit the :focus this week.

**Comment by toolness** [comment]
Awesome, thanks claudio!

File: app/static/sass/_components.scss [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    font-weight: 400;
+    letter-spacing: 1px;
+    color: $white;
+    text-transform: uppercase;
+    position: relative;
+    border-radius: 0;
+    box-shadow: inset 0 0 3px $gray-10;
+
+    option:before {
+        content: "claudio"
```
Comment (toolness) [inline app/static/sass/_components.scss]: Um, is this supposed to be here @claudioccm?

Comment (toolness) [inline app/static/sass/_components.scss]: lol I am assuming this is unintended:

**Comment by toolness** [comment]
Just added the following concern to the issue description, @claudioccm:

> The close button on flash messages is currently a `<div>`, but it should really be a `<button>` for full accessibility (an alternative is for us to add a bunch of stuff like keyboard event handlers and `tabindex` attributes, but that's quite a hassle). The same goes for any other elements that have the functionality of buttons, even if they don't look like them.

**Pull request merged**